src/System.Runtime.Extensions/src/System/IO/StringReader.cs

```diff
@@ -117,6 +118,35 @@ public override int Read(char[] buffer, int index, int count)
             return n;
         }
 
+        public override int Read(Span<char> destination)
+        {
+            if (GetType() != typeof(StringReader))
```

Comment:
Do we want to add the comment on why we do the check for derived classes?

Reply:
Sure, I'll add comments.